src/node_env_var.cc

```diff
@@ -121,7 +121,7 @@ void RealEnvStore::Set(Isolate* isolate,
   node::Utf8Value val(isolate, value);
 
 #ifdef _WIN32
-  if (key[0] == L'=') return;
+  if (key.length() > 0 && key[0] == L'=') return;
```

Comment:
The change itself LGTM but this points to a deeper issue with `Utf8Value` and `MaybeStackBuffer`. They zero-terminate the string buffer, therefore `key[0]` is a valid index, but it's rejected by the latter's `operator[]` due to an overly strict assert.

You can't use iterator-like patterns either for the same reason even though it's legal C++:
```c++
size_t n = key.length();
char* p = &key[0]
char* pe = &key[n];  // asserts
while (p != pe) { /* ... */ }
```
I'm a little undecided whether it's better to relax the check or just live with it and work around it like you're doing here.

Aside: it shouldn't compare against a wide character (the `L'='`) but that makes no material difference here.

Reply:
Hi @bnoordhuis,

Thank you for your thoughts.  I understand your point regarding `MaybeStackBuffer`.  It seems best to address the larger issue rather than work around it.  I removed the workaround and relaxed the assertion.